internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,26 @@
+package buffer
+
+import "io"
+
+// WriteBuffer is an interface that buffers the data to be written to GCS during
+// the write flow.
+// WriteBuffer is used only in create new file flow with sequential writes and
+// at any point in time, only 2x of the configured buffer size is stored in the
+// write buffer.
+type WriteBuffer interface {
+	io.Reader
```

Comment:
still not clear to me why we need this. add it when go-client library methods are added/